examples/dyldinfo.rs

```diff
@@ -125,12 +126,18 @@ fn print_multi_arch(
         if let Some((cputype, _)) = mach::constants::cputype::get_arch_from_flag(&arch) {
             for bin in multi_arch.into_iter() {
                 match bin {
-                    Ok(bin) => {
+                    Ok(MultiArchEntry::MachO(bin)) => {
```

Comment:
the name `MultiArchEntry`, is this some kind of "prior art", e.g., did you see this in LLVM, or in otool/lipo references? Wondering about different possible names, for example, I'll show enum expanded with the Macho variant:

1. `Entry::MachO`
2. `MultiBin::MachO`
3. `MultiEntry::MachO`
4. `ArchEntry::MachO`
5. `BinaryKind::MachO`
6. `EntryKind::MachO`

I personally like what you have, and 1, 4, and 5. Do you have any thoughts on names here?

This is not the most interesting to talk about names here, but since this will become a public API, will be good to take small time to think about alternatives :)

Reply:
I just made it up. Having a look around, I can't find reference to them in LLVM. In `lipo`, the man pages say this:

>  lipo supports a number of commands for creating universal files from single-architecture files, extracting single-architecture files from universal files, and displaying architecture information.

So perhaps `SingleArch`, then you could get a collection of `SingleArch`'s from a `MultiArch`?

Another alternative would be something to do with "member" because multi-arch binaries are expanded archives and archives are made up of "members" (or at least that's the language `ar` uses).

Of the options you listed, I quite like 4. I think that the name indicating that this is a particular arch is good.